### DISCUSSION TITLE
Airflow 2.5.3 (docker) DAG Import Errors: cron_descriptor cannot find en_US.mo

### DISCUSSION THREAD

**Discussion by andrecastro0o** [apache/airflow#31715] [Q&A]
Hi I have tried to upgrade airflow locally, and am running into a strange error connection with the cron_descriptor library

```Broken DAG: [/opt/airflow/dags/file_transfers/ft_DAVIS002PD.py] Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/cron_descriptor/GetText.py", line 48, in __init__
  File "/home/airflow/.local/lib/python3.8/site-packages/cron_descriptor/GetText.py", line 63, in load_locale
FileNotFoundError: [Errno 2] No such file or directory: '/home/airflow/.local/lib/python3.8/site-packages/cron_descriptor/locale/en_US.mo'
```

If i list the files in that dir ` docker exec ruisdael-airflow-airflow-webserver-1 ls /home/airflow/.local/lib/python3.8/site-packages/cron_descriptor/locale/` is present
`-rw-rw-r-- 1 airflow root 3719 Mar 31 22:58 en_US.mo` 

what might be the cause for the error

![Screenshot 2023-06-05 at 16-37-09 file_transfer_DAVIS002_2_PD - Grid - Airflow](https://github.com/apache/airflow/assets/1690557/2abb847f-21b3-4e09-bfa3-dd073816eae0)

**Comment by potiuk** [comment]
Looks like your system has broken locale. Example here, but you should search google for resolution of your problem : https://askubuntu.com/questions/599808/cannot-set-lc-ctype-to-default-locale-no-such-file-or-directory

**Reply by andrecastro0o** [reply]
thanks @potiuk. is that the case? I am unsure, since I am running airflow with docker containers. By system you mean the host or one of the the containers

**Reply by potiuk** [reply]
My wild guess is containers - the link with similar issu points out to some locale missing , so maybe you need to build/extend your containers to add something that is missing there.

**Reply by andrecastro0o** [reply]
I looked at what was happening is that airflow was looking '/home/airflow/.local/lib/python3.8/site-packages/cron_descriptor/locale/en_US.mo' and inside the airflow web container, python3.7 was used, hence `/home/airflow/.local/lib/python3.8/` did not exist, but `/home/airflow/.local/lib/python3.7/`

I have addressed the issue by specifying what airflow image version, I want to use in the docker-compose file

```yaml
  image: apache/airflow:2.6.1-python3.10
```
 

(I believe the issue can be closed. Thank you @potiuk )

**Reply by potiuk** [reply]
Wild guess - naybe you  had some old .pyc files. If for example you have shared volumes and you try to run your DAG with Python 3.8, Python might generate a .pyc file using the Python 3.8 interpreter and it might contain code to import in the other or maybe you had bad PYTHONPATH passed somewhere. Or maybe the images were actually different in different components. you could have old version of Python 3.7 image runnig in one docker engine and Python 3.8. If you for example changed the version of python in your image but did not change name of it, it could be that two engines could have different versions of image with the same name.

There could be many reasons, and I generally recommend always uniquely name the image tha you use.
